src/tools/cephfs/top/cephfs-top

```diff
@@ -0,0 +1,294 @@
+            coord = x_coord_map[item]
+            if item == FS_TOP_MAIN_WINDOW_COL_CLIENT_ID:
+                self.mainw.addstr(y_coord, coord[0], client_id.split('.')[1])
+            elif item == FS_TOP_MAIN_WINDOW_COL_MNTPT:
+                self.mainw.addstr(y_coord, coord[0], client_meta[CLIENT_METADATA_MOUNT_POINT_KEY])
+            elif item == FS_TOP_MAIN_WINDOW_COL_MNT_ROOT:
+                self.mainw.addstr(y_coord, coord[0], client_meta[CLIENT_METADATA_MOUNT_ROOT_KEY])
+        cidx = 0
+        for item in counters:
+            coord = x_coord_map[item]
+            m = metrics[cidx]
+            typ = MAIN_WINDOW_TOP_LINE_METRICS[MGR_STATS_COUNTERS[cidx]]
+            if item.lower() in client_meta['valid_metrics']:
+                if typ == MetricType.METRIC_TYPE_PERCENTAGE:
+                    self.mainw.addstr(y_coord, coord[0], f'{calc_perc(m)}')
+                elif typ == MetricType.METRIC_TYPE_LATENCY:
+                    self.mainw.addstr(y_coord, coord[0], f'{calc_lat(m)}')
+            else:
+                self.mainw.addstr(y_coord, coord[0], "N/A")
+            cidx += 1
+
+    def refresh_clients(self, x_coord_map, stats_json):
+        counters = [m.upper() for m in stats_json[GLOBAL_COUNTERS_KEY]]
+        y_coord = 0
+        for client_id, metrics in stats_json[GLOBAL_METRICS_KEY].items():
+            self.refresh_client(client_id,
+                                metrics,
+                                counters,
+                                stats_json[CLIENT_METADATA_KEY][client_id],
+                                x_coord_map,
+                                y_coord)
+            y_coord += 1
+
+    def refresh_main_window(self, x_coord_map, stats_json):
+        self.refresh_clients(x_coord_map, stats_json)
+
+    def refresh_header(self, stats_json):
+        if not stats_json['version'] == FS_TOP_SUPPORTED_VER:
+            self.header.addstr(0, 0, 'perf stats version mismatch!')
+            return False
+        client_metadata = stats_json[CLIENT_METADATA_KEY]
+        num_clients = len(client_metadata)
+        num_mounts = len([client for client, metadata in client_metadata.items() if not
+                          metadata[CLIENT_METADATA_MOUNT_POINT_KEY] == 'N/A'])
+        now = datetime.now().strftime('%H:%M:%S')
+        self.header.addstr(0, 0, FS_TOP_VERSION_HEADER_FMT.format(
+            prog_name=FS_TOP_PROG_STR, prog_ver=FS_TOP_PROG_VER, now=now),
+            curses.A_STANDOUT | curses.A_BOLD)
+        self.header.addstr(1, 0, FS_TOP_CLIENT_HEADER_FMT.format(
+            num_clients=num_clients, num_mounts=num_mounts, num_rest=num_clients - num_mounts))
+        return True
+
+    def display(self, _):
+        x_coord_map = self.refresh_top_line_and_build_coord()
+        self.topl.refresh()
+        while not self.stop:
+            stats_json = self.perf_stats_query()
+            self.header.clear()
+            self.mainw.clear()
```

Comment:
I think these `clear()` calls are what's causing the blinking every second. Is that right? There should be a way to fix that...

Reply:
Probably, but I'm not sure. @mchangir any idea?